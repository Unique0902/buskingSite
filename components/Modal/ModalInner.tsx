import React, { ReactNode, useEffect, useRef } from 'react';

type Props = {
  children: ReactNode;
  handleClickOther: () => void;
};

const ModalInner = ({ children, handleClickOther }: Props) => {
  const modalRef = useRef<HTMLDivElement>(null);
  useEffect(() => {
    const handleClickOutside = (event: MouseEvent) => {
      if (
        modalRef.current &&
        !modalRef.current.contains(event.target as Node)
      ) {
        handleClickOther();
      }
    };
    document.addEventListener('mousedown', handleClickOutside);
    return () => {
      document.removeEventListener('mousedown', handleClickOutside);
    };
  }, [modalRef, handleClickOther]);
  return (
    <section className='fixed top-0 left-0 flex flex-row items-center justify-center w-screen h-screen gap-4'>
      <div
        ref={modalRef}
        className='z-20 w-1/2 text-black bg-white opacity-100 rounded-2xl max-lg:rounded-none dark:bg-slate-500 h-5/6 max-lg:w-full max-lg:h-full'
      >
        {children}
      </div>
      <div className='absolute z-10 w-screen h-screen bg-black opacity-50'></div>
    </section>
  );
};

export default ModalInner;
