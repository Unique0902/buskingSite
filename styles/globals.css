@tailwind base;
@tailwind components;
@tailwind utilities;
/* TODO: 글꼴 불러오는 로딩 줄이거나 없애기 */
* {
  box-sizing: border-box;
  font-family: 'Pretendard';
}

@font-face {
  font-family: 'Pretendard';
  font-weight: 45 920;
  font-style: normal;
  font-display: swap;
  src: url('../assets/fonts/PretendardVariable.woff2')
    format('woff2-variations');
}

li {
  list-style: none;
}
body {
  margin: 0;
  @apply text-black dark:text-white dark:bg-slate-600;
}
p,
h1,
h2 {
  margin: 0;
}
ul {
  padding: 0;
}
