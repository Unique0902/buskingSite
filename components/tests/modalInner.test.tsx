import { screen, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import { expect, describe, it, jest, afterEach } from '@jest/globals';
import ModalInner from '../Modal/ModalInner';

describe('ModalInner component test', () => {
  const handleClickOther = jest.fn();

  afterEach(() => handleClickOther.mockReset());

  it('renders correctly', () => {
    const component = renderer.create(
      <ModalInner handleClickOther={handleClickOther}>
        <div>inner</div>
      </ModalInner>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('click inner and handleClickOther not execute', async () => {
    render(
      <>
        <ModalInner handleClickOther={handleClickOther}>
          <div>inner</div>
        </ModalInner>
        <div>outer</div>
      </>
    );

    const innerBox = screen.getByText('inner');
    await userEvent.click(innerBox);
    expect(handleClickOther).not.toHaveBeenCalled();
  });

  it('click outer and handleClickOther execute', async () => {
    render(
      <>
        <ModalInner handleClickOther={handleClickOther}>
          <div>inner</div>
        </ModalInner>
        <div>outer</div>
      </>
    );

    const outerBox = screen.getByText('outer');
    await userEvent.click(outerBox);
    expect(handleClickOther).toHaveBeenCalled();
  });
});
