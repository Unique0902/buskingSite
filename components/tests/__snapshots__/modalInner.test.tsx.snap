// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalInner component test renders correctly 1`] = `
<section
  className="fixed top-0 left-0 flex flex-row items-center justify-center w-screen h-screen gap-4"
>
  <div
    className="z-20 w-1/2 text-black bg-white opacity-100 rounded-2xl max-lg:rounded-none dark:bg-slate-500 h-5/6 max-lg:w-full max-lg:h-full"
  >
    <div>
      inner
    </div>
  </div>
  <div
    className="absolute z-10 w-screen h-screen bg-black opacity-50"
  />
</section>
`;
